cpu/cc2538/Makefile.include

```diff
@@ -1 +1,56 @@
+
+# Set ROM and RAM lengths according to CPU model
+ifneq (,$(filter cc2538nf11,$(CPU_MODEL)))
+  ifneq (,$(filter riotboot,$(FEATURES_USED)))
+    # If using riotboot loose 2 pages to not right over CCA section and
+    # keep page parity
+    ROM_LEN ?= 124K
+  else
+    ROM_LEN ?= 128K
+  endif
+  RAM_LEN ?= 16K
+endif
+ifneq (,$(filter cc2538nf23,$(CPU_MODEL)))
+  ifneq (,$(filter riotboot,$(FEATURES_USED)))
+    # If using riotboot loose 2 pages to not right over CCA section and
+    # keep page parity
+    ROM_LEN ?= 252K
```

Comment:
Shouldn't we *always* prevent overwriting the last page then? So unconditionally subtracting 2k?
And the 2k in the start, aren't they taken into account if you set `ROM_OFFSET`? 

That way, we wouldn't need to copy & paste this for every model

Reply:
On a non riotboot application there is no issue with overriding the CCA section.

> Shouldn't we always prevent overwriting the last page then? So unconditionally subtracting 2k?
> And the 2k in the start, aren't they taken into account if you set ROM_OFFSET?

cc2538 pages are 2k long, the bootloader takes 4k so two pages which keeps page parity. But here if we remove only the cca we would loose parity, so I remove an extra page.

Reply:
I don't understand what that page parity is about

Reply:
So how about 

```make
ifneq (,$(filter riotboot,$(FEATURES_USED)))
  CCA_SIZE := 2048
else
  CCA_SIZE := 0
endif

FLASHSIZE := $(shell echo $$(( $(FLASHSIZE) - 2 * $(CCA_SIZE) )) )
```

Reply:
`FLASHSIZE` is not the inportant define here though, its `ROM_LEN` since that one then gets passed to the linker:

```
  LINKFLAGS += $(LINKFLAGPREFIX)--defsym=_rom_length=$(ROM_LEN)
  LINKFLAGS += $(LINKFLAGPREFIX)--defsym=_ram_length=$(RAM_LEN)
```

I'll take a second look and try to improve though.

Reply:
I adapted to fir your suggestion, let me know what you think @benpicco